File: src/main/java/projetoSoftware/com/mercado/model/Venda.java
package projetoSoftware.com.mercado.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import projetoSoftware.com.mercado.enumerado.statusVenda;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;

@Data
@Entity
@Table(name = "venda")
@AllArgsConstructor
@NoArgsConstructor
public class Venda {
    @Column(name = "identificador")
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Integer identificador;

    @Column(name="vendedor_id")
    private String vendedor;

    @Column(name = "cliente_id")
    private String cliente;

    @Column(name= "formaPagamento")
    private String formaPagamento;

    @Column(name = "totalAPagar")
    private int totalAPagar;

    @Column(name = "data")
    private Date data;

    @Column(name = "status")
    private Enum<statusVenda> status;


}

File: src/Modelos/Venda.js
export default class Venda {

    constructor(vendedor, cliente, produtos, formaDePagamento) {
        this._vendedor = vendedor;
        this._cliente = cliente;
        this._produtos = produtos;
        this._formaDePagamento = formaDePagamento
    }

    get vendedor() {
        return this._vendedor;
    };

    set vendedor(vendedor) {
        return this._vendedor = vendedor;
    }

    get cliente() {
        return this._cliente;
    }

    set cliente(cliente) {
        return this._cliente = cliente;
    }

    get produtos() {
        return this._produtos;
    };

    set produtos(produtos) {
        return this._produtos = produtos;
    }

    get cliente() {
        return this._formaDePagamento;
    }

    set cliente(formaDePagamento) {
        return this._formaDePagamento = formaDePagamento;
    }
    
}